src/components/AdvancedSelect.vue

```diff
@@ -534,6 +523,34 @@ export default {
       }
       return null;
     },
+    filter_options(options) {
+      return clone(options).filter(this.recursive_match);
+    },
+    get_linear_options(options) {
+      return options.reduce((f, o) => {
+        if (o.options) {
+          // push the header
+          f.push({
+            header: o.label,
+          });
+          f.push(...o.options.map(opt => Object.assign({}, opt, { parentHeader: o.label, selected: this.valueIsSelected(o.value) })));
+        } else {
+          // it's an item without group, push it to the list
+          f.push(Object.assign({}, o, { selected: this.valueIsSelected(o.value) }));
+        }
+        return f;
+      }, []);
+    },
+    recursive_match(o) {
+      if (this.optionMatch(o)) {
+        return true;
+      }
+
+      if (o.options) {
+        return (o.options = o.options.filter(this.recursive_match)).length;
```

Comment:
The method should return `true` or `false` and that's it. No side effects. Find another way to `reduce` this.

Reply:
Updated